Code/Framework/AzFramework/Tests/Spawnable/SpawnableEntitiesManagerTests.cpp

```diff
@@ -366,33 +542,144 @@ namespace UnitTest
+              Spawnable::EntityAliasType::Replace },
+            &target);
+
+        size_t spawnedEntitiesCount = 0;
+        bool allReplaced = true;
+        auto callback = [&spawnedEntitiesCount, &allReplaced](
+            AzFramework::EntitySpawnTicket::Id, AzFramework::SpawnableConstEntityContainerView entities)
+        {
+            spawnedEntitiesCount += entities.size();
+            allReplaced = AreAllEntitiesReplaced(entities);
+        };
+        AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+        optionalArgs.m_completionCallback = AZStd::move(callback);
+        m_manager->SpawnAllEntities(*m_ticket, AZStd::move(optionalArgs));
+        m_manager->ProcessQueue(AzFramework::SpawnableEntitiesManager::CommandQueuePriority::Regular);
+
+        EXPECT_EQ(4, spawnedEntitiesCount);
+        EXPECT_TRUE(allReplaced);
+    }
+
+    TEST_F(SpawnableEntitiesManagerTest, SpawnAllEntities_AllAliasesWithAdditional_SourceAndTargetComponentsMerged)
+    {
+        using namespace AzFramework;
+        static constexpr size_t NumEntities = 4;
+        FillSpawnable(NumEntities);
+        AZ::Data::Asset<Spawnable> target = CreateTargetSpawnable(4);
+        InsertEntityAliases<4>(
+            { 0, 1, 2, 3 }, { 0, 1, 2, 3 },
+            { Spawnable::EntityAliasType::Additional, Spawnable::EntityAliasType::Additional, Spawnable::EntityAliasType::Additional,
+              Spawnable::EntityAliasType::Additional },
+            &target);
+
+        size_t spawnedEntitiesCount = 0;
+        bool allAdded = true;
+        auto callback = [&spawnedEntitiesCount, &allAdded](
+            AzFramework::EntitySpawnTicket::Id, AzFramework::SpawnableConstEntityContainerView entities)
+        {
+            spawnedEntitiesCount += entities.size();
+            allAdded = IsEveryOtherEntityAReplacement(entities);
+        };
+        AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+        optionalArgs.m_completionCallback = AZStd::move(callback);
+        m_manager->SpawnAllEntities(*m_ticket, AZStd::move(optionalArgs));
         m_manager->ProcessQueue(AzFramework::SpawnableEntitiesManager::CommandQueuePriority::Regular);
+
+        EXPECT_EQ(8, spawnedEntitiesCount);
+        EXPECT_TRUE(allAdded);
     }
 
+    TEST_F(SpawnableEntitiesManagerTest, SpawnAllEntities_AllAliasesWithMerge_SourceAndTargetComponentsMerged)
+    {
+        using namespace AzFramework;
+        static constexpr size_t NumEntities = 4;
+        FillSpawnable(NumEntities);
+        AZ::Data::Asset<Spawnable> target = CreateTargetSpawnable(4);
+        InsertEntityAliases<4>(
+            { 0, 1, 2, 3 }, { 0, 1, 2, 3 },
+            { Spawnable::EntityAliasType::Merge, Spawnable::EntityAliasType::Merge, Spawnable::EntityAliasType::Merge,
+              Spawnable::EntityAliasType::Merge },
+            &target);
+
+        size_t spawnedEntitiesCount = 0;
+        bool allMerged = true;
```

Comment:
repeated : recommend changing the default to 'false'. That way, later when you do 'EXPECT_TRUE(allMerged)', we can be sure that the lambda actually worked and flipped the bool to true. Otherwise, if the call to SEM fails, the bool can result in a false positive.